### PR TITLE
[IOTDB-3547] AddConfigNode support procedure

### PR DESCRIPTION
**Pull request by wangchao316**
1. The system can be started concurrently.
2. Supports leader switchover.
https://apache-iotdb.feishu.cn/docs/doccniPRgxESvXlekL0PuSiHolf#HMudTM

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/AddConfigNodeProcedure.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  }
+
+  @Override
+  protected void rollbackState(ConfigNodeProcedureEnv env, AddConfigNodeState state)
+      throws IOException, InterruptedException {
+    switch (state) {
+      case ADD_CONSENSUS_GROUP:
+      case ADD_PEER:
+        LOG.info("Rollback preDeleted:{}", tConfigNodeLocation);
+
```
Comment (cmlmakahts) [inline confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/AddConfigNodeProcedure.java:103]: Log msg should be fixed

Comment (wangchao316) [reply confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/AddConfigNodeProcedure.java]: fixed

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConsensusManager.java [hunk 2]
```diff
@@ -102,29 +81,28 @@ private void setConsensusLayer(PartitionRegionStateMachine stateMachine) throws
                             conf.getConfigNodeConsensusProtocolClass())));
     consensusImpl.start();
 
-    // Build consensus group from iotdb-confignode.properties
-    LOGGER.info("Set ConfigNode consensus group {}...", conf.getConfigNodeList());
+    addConsensusGroup(conf.getConfigNodeList());
```
[the remark this answers is not in the document]
Comment (wangchao316) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ConsensusManager.java]: fixed

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConsensusManager.java [hunk 3]
```diff
@@ -158,7 +136,29 @@ public boolean isLeader() {
     return consensusImpl.isLeader(consensusGroupId);
   }
 
+  public Peer getLeader() {
+    return consensusImpl.getLeader(consensusGroupId);
+  }
+
```
[the remark this answers is not in the document]
Comment (wangchao316) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ConsensusManager.java]: fixed

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConsensusManager.java [hunk 4]
```diff
@@ -102,29 +81,28 @@ private void setConsensusLayer(PartitionRegionStateMachine stateMachine) throws
...
+  /**
+   * execute in new node
+   *
+   * @param configNodeLocations all config node
+   */
+  public void addConsensusGroup(List<TConfigNodeLocation> configNodeLocations) {
+    if (configNodeLocations.size() == 0) {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (wangchao316) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ConsensusManager.java]: fixed

**Review by CRZbulabula — APPROVED** [review]
LGTM~

**Comment by cmlmakahts** [comment]
LGTM

**Review by cmlmakahts — APPROVED** [review]
LGTM

**Pull request merged**